src/federation/observer.rs

```diff
@@ -75,13 +75,44 @@ impl FederationObserver {
     }
 
     async fn setup_schema(&self) -> anyhow::Result<()> {
-        self.connection()
-            .await?
-            .batch_execute(include_str!(concat!(
-                env!("CARGO_MANIFEST_DIR"),
-                "/schema/v0.sql"
-            )))
-            .await?;
+        let schema_version = query_value::<i32>(
+            &self.connection().await?,
+            // language=postgresql
+            "SELECT
+                    CASE
+                        WHEN EXISTS (
+                            SELECT FROM information_schema.tables
+                            WHERE table_schema = 'public'
+                            AND table_name = 'schema_version'
+                        )
+                        THEN (SELECT COALESCE(MAX(version), 0) FROM schema_version)
+                        ELSE 0
+                    END AS max_version;",
```

Comment:
Seemed to fail for me when deployed, but was a very messy setup

Reply:
I'm also bumping into issues. This fixes on my end, but feel free to make tweaks

https://github.com/elsirion/fedimint-observer/pull/30